src/krb5_common_plug.c

```diff
@@ -268,13 +268,10 @@ int des_string_to_key_shishi(char *string, size_t stringlen,
 	n_s = stringlen + saltlen;
 	if ((n_s % 8) != 0)
 		n_s += 8 - n_s % 8;
-	s = malloc(n_s);
-	s_copy = malloc(n_s);
 	memset(s, 0, n_s);
 	memcpy(s, string, stringlen);
 	if (saltlen > 0)
 		memcpy(s + stringlen, salt, saltlen);
-	memset(s + stringlen + saltlen, 0, n_s - stringlen - saltlen);
 	memset(tempkey, 0, sizeof(tempkey));
 	memcpy(s_copy, s, n_s);  // this is required as the following loop has an unknown bug which corrupts the last block of "s"
```

Comment:
Unknown bug? Per definition it's known - and you should bloody fix it instead of work around it!

Reply:
This code is not mine. It is borrowed from the Shishi project. If this bug was known to them, they would have fixed it five years ago! I will update the comment so that the situation is more clear.

Reply:
Ahaaa, makes more sense now.